### PR TITLE
Dash/oren/global bgs cassette cleanup

### PR DESCRIPTION
**Pull request by nihil2501**
# TL;DR
No more need to manually accommodate BGS cassettes when running the test suite locally, anywhere in `vets-api`!

## Problem
Previously, one could experience test failures for any test that was recorded with a hardcoded BGS base URL if the value in their local settings did not match that in the cassette. In this case, in order to get the test to pass, they would have to take the manual step of updating the value in their local settings.

## Solution
The solution is for these cassettes to instead use a value that is exactly identical to the dynamic value that BGS base URL actually takes (`Settings.bgs.url`). The [cassette placeholder feature](https://benoittgt.github.io/vcr/#/configuration/filter_sensitive_data?id=filter-sensitive-data) of VCR is exactly what is needed to accomplish this automatically, and we recently added such a placeholder `<BGS_BASE_URL>` value [here](https://github.com/department-of-veterans-affairs/vets-api/blob/d3a244cec4249e4d94eabb8070a8f3c8ff11b485/spec/support/vcr.rb#L12).

This PR replaces the hardcoded value with the dynamic value everywhere that it occurs.

## Aside
I was also looking to clean up cassettes "orphaned" by [this recent change](https://github.com/department-of-veterans-affairs/vets-api/commit/7b7ffafe9ad5cb8d319fe9db6aad60e6eff7f91a), but decided that there isn't necessarily such a cut and dry notion as that when it comes to how cassettes have been used in this codebase, and the change isn't so significant.

### CODE REVIEW

**Review by rmtolmach — APPROVED** [review]
Thanks for explaining where `BGS_BASE_URL` was set. 

I clicked "Load diff" for 75% of these cassettes and they look good. 😅 Trusting that the other 25% are also 👍

**Review by kpethtel — APPROVED** [review]
Mobile changes look good.
